modules/integration/src/test/scala/scala/cli/integration/CompileTestDefinitions.scala

```diff
@@ -57,40 +88,47 @@ abstract class CompileTestDefinitions(val scalaVersionOpt: Option[String])
     }
   }
 
-  test("test scope") {
+  def checkIfCompileOutputIsCoppied(baseName: String, output: os.Path): Unit =
+    expect(os.list(output).map(_.relativeTo(output)) == compileFilesExtensions.map(ext =>
+      os.rel / s"$baseName$ext"
+    ))
+
+  test("copy compile output") {
     val inputs = TestInputs(
       Seq(
-        os.rel / "Main.scala" ->
-          """//> using lib "com.lihaoyi::utest:0.7.10"
-            |
-            |object Main {
-            |  val err = utest.compileError("pprint.log(2)")
-            |  def message = "Hello from " + "tests"
-            |  def main(args: Array[String]): Unit = {
-            |    println(message)
-            |    println(err)
-            |  }
-            |}
-            |""".stripMargin,
-        os.rel / "Tests.scala" ->
-          """//> using lib "com.lihaoyi::pprint:0.6.6"
-            |//> using target.scope "test"
-            |
-            |import utest._
-            |
-            |object Tests extends TestSuite {
-            |  val tests = Tests {
-            |    test("message") {
-            |      assert(Main.message.startsWith("Hello"))
-            |    }
-            |  }
-            |}
-            |""".stripMargin
+        mainInput,
+        testInput
+      )
+    )
+    inputs.fromRoot { root =>
+      val tempOutput    = root / "output"
+      val outputOptions = Seq("--output", tempOutput.toString)
+      val cmd           = os.proc(TestUtil.cli, "compile", outputOptions, extraOptions, ".")
+      cmd.call(cwd = root, check = false, stderr = os.Pipe, mergeErrIntoOut = true)
```

Comment:
No need to pass `check = false` and all, it seems:
```suggestion
      os.proc(TestUtil.cli, "compile", outputOptions, extraOptions, ".").call(cwd = root)
```